tests/server/activity-event.js

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'intern!object',
+  'intern/chai!assert',
+  'intern/dojo/node!../../server/lib/activity-event',
+  'intern/dojo/node!bluebird',
+  'intern/dojo/node!path',
+  'intern/dojo/node!sinon',
+], function (registerSuite, assert, activityEvent, Promise, path, sinon) {
+  var write;
+
+  registerSuite({
+    name: 'activity-event',
+
+    'interface is correct': function () {
+      assert.isFunction(activityEvent);
+      assert.lengthOf(activityEvent, 3);
+    },
+
+    'call activityEvent with service, without client_id': {
+      setup: function () {
+        write = process.stderr.write;
+        process.stderr.write = sinon.spy();
+        activityEvent('mock event', { a: 'b', c: 'd' }, {
+          headers: { 'user-agent': 'foo' },
+          originalUrl: 'bar',
+          query: {
+            context: 'mock context',
+            entrypoint: 'mock entrypoint',
+            migration: 'mock migration',
+            service: 'mock service',
+            /*eslint-disable camelcase*/
+            utm_campaign: 'mock utm_campaign',
+            utm_content: 'mock utm_content',
+            utm_medium: 'mock utm_medium',
+            utm_source: 'mock utm_source',
+            utm_term: 'mock utm_term',
+            /*eslint-enable camelcase*/
+            zignore: 'ignore me'
+          }
+        });
+        return new Promise(function (resolve) {
+          setImmediate(function () {
```

Comment:
Maybe there is a better way to do this.

The `setImmediate` is so that the setup doesn't complete until `activityEvent` finishes. My thinking was that `setImmediate` handlers are called in order, so the simplest way to wait for it to finish seemed like resolving the returned promise in a `setImmediate` of its own.

Make sense? Is that a stupid way of doing it?